tests/unit_tests/cuda/detector_cuda_kernel.cu

```diff
@@ -105,4 +105,42 @@ void detector_test(
     DETRAY_CUDA_ERROR_CHECK(cudaDeviceSynchronize());
 }
 
+// cuda kernel to test enumeration
+__global__ void enumerate_test_kernel(
+    detector_view<detector_host_t> det_data,
+    vecmem::data::jagged_vector_view<surface_t> surfaces_data) {
+
+    // Get detector in device
+    detector_device_t detector(det_data);
+
+    // Get surface vector per volume
+    vecmem::device_vector<surface_t> surfaces_device(
+        surfaces_data.m_ptr[threadIdx.x]);
```

Comment:
@krasznaa Is there any risk in defining `device_vector` from the element of `jagged_vector_view`?

Reply:
I don't like this... :frowning: This would tie our hands a bit too much with how we could re-design the internals of these classes. In general, you shouldn't ever use the members of these classes.

The "clean" setup for this vector should look like:

```c++
vecmem::jagged_device_vector<surface_t> all_surfaces(surfaces_data);
vecmem::device_vector<surface_t> surfaces_device = all_surfaces.at(threadIdx.x);
```

As an aside: To resemble [std::vector](https://en.cppreference.com/w/cpp/container/vector) a bit more, the operators on the device vectors are "not checked" at all. So giving them an invalid index is undefined behaviour. (Mostly something that you'd need to use [cuda-memcheck](https://docs.nvidia.com/cuda/cuda-memcheck/index.html) to catch.) The `at(...)` functions on the other hand **are** checked. But only with an assertion. Since in device code exceptions would not work.

So while the `[]` operators are there, we should use `at(...)` calls in most places in our code to help with debugging the code. In optimised builds the use of `at(...)` will not have any negatives, unlike `std::vector::at(...)`.

Reply:
OK. Thanks for the suggestion.
I changed to use `at()`